pyresample/ewa/dask_ewa.py

```diff
@@ -0,0 +1,528 @@
+how much data is being worked on at any one time. This limits how much
+data is being held in memory at any one time. In cases where not all input
+array chunks will be used this implementation should avoid loading/computing
+the data for that chunk. In cases where not all output chunks have data in
+them, this implementation should avoid unnecessary array creation and memory
+usage until necessary.
+
+"""
+import math
+import logging
+from functools import partial
+
+from pyresample.geometry import SwathDefinition
+from pyresample.resampler import BaseResampler, update_resampled_coords
+from pyresample.ewa import ll2cr
+from pyresample.ewa._fornav import (fornav_weights_and_sums_wrapper,
+                                    write_grid_image_single)
+
+import dask
+import dask.array as da
+from dask.array.core import normalize_chunks
+from dask.highlevelgraph import HighLevelGraph
+import numpy as np
+
+try:
+    import xarray as xr
+except ImportError:
+    # only used for some use cases
+    xr = None
+
+logger = logging.getLogger(__name__)
+
+
+def _call_ll2cr(lons, lats, target_geo_def):
+    """Wrap ll2cr() for handling dask delayed calls better."""
+    new_src = SwathDefinition(lons, lats)
+    swath_points_in_grid, cols, rows = ll2cr(new_src, target_geo_def)
+    if swath_points_in_grid == 0:
+        return (lons.shape, np.nan, lons.dtype), (lats.shape, np.nan, lats.dtype)
+    return np.stack([cols, rows], axis=0)
+
+
+def call_ll2cr(lons, lats, target_geo_def):
+    res = da.map_blocks(_call_ll2cr, lons, lats,
+                        target_geo_def,
+                        dtype=lons.dtype)
+    return res
+
+
+def _delayed_fornav(ll2cr_result, target_geo_def, y_slice, x_slice, data, fill_value, kwargs):
+    # Adjust cols and rows for this sub-area
+    subdef = target_geo_def[y_slice, x_slice]
+    weights_dtype = np.float32
+    accums_dtype = np.float32
+    empty_weights = (subdef.shape, 0, weights_dtype)
+    empty_accums = (subdef.shape, 0, accums_dtype)
+
+    # Empty ll2cr results: ((shape, fill, dtype), (shape, fill, dtype))
+    if isinstance(ll2cr_result[0], tuple):
+        # this source data doesn't fit in the target area at all
+        return empty_weights, empty_accums
+    cols = ll2cr_result[0]
+    rows = ll2cr_result[1]
+    if x_slice.start != 0:
+        cols = cols - x_slice.start
+    if y_slice.start != 0:
+        rows = rows - y_slice.start
+    weights = np.zeros(subdef.shape, dtype=weights_dtype)
+    accums = np.zeros(subdef.shape, dtype=accums_dtype)
+    try:
+        got_points = fornav_weights_and_sums_wrapper(
+            cols, rows, data, weights, accums, fill_value, fill_value,
+            **kwargs)
+    except RuntimeError:
+        return empty_weights, empty_accums
+    if not got_points:
+        return empty_weights, empty_accums
+    return weights, accums
+
+
+def chunk_callable(x_chunk, axis, keepdims, **kwargs):
```

Comment:
Should this be a private function?

Reply:
Done. I also updated the other functions in this module to be `_` prefixed.